src/index.js

```diff
@@ -0,0 +1,122 @@
+    constructor(config = {}) {
+        const chluIpfsConfig = get(config, 'chluIpfsConfig', {})
+        const directory = path.join(process.env.HOME, '.chlu-publish')
+        if (!chluIpfsConfig.directory) chluIpfsConfig.directory = directory
+        this.chluIpfs = get(config, 'chluIpfs', null)
+        if (!this.chluIpfs) this.chluIpfs = new ChluIPFS(chluIpfsConfig)
+        this.port = get(config, 'port', 3006)
+        this.logger = get(config, 'logger', this.chluIpfs.logger)
+        this.prepareAPI()
+        this.log = msg => this.logger.debug(`[API] ${msg}`)
+    }
+
+    async start() {
+        await this.chluIpfs.start()
+        this.log('Starting HTTP Server')
+        await new Promise(resolve => this.api.listen(this.port, resolve))
+        this.log(`Started HTTP Server on port ${this.port}`)
+    }
+
+    async stop() {
+        await this.chluIpfs.stop()
+    }
+
+    prepareAPI() {
+        this.api = express()
+        this.api.use(cors())
+        this.api.use(bodyParser.json())
+        this.api.get('/', (req, res) => res.send('Chlu API Publish').end())
+        const apiv1 = this.prepareAPIV1()
+        this.api.use('/api/v1', apiv1)
+    }
+
+    prepareAPIV1() {
+        const api = express()
+
+        api.get('/id', async (req, res) => {
+            try {
+                this.log('GET ID => ...')
+                await this.chluIpfs.waitUntilReady()
+                const did = this.chluIpfs.didIpfsHelper.didId
+                this.log(`GET ID => ${did}`)
+                res.json({ did })
+            } catch (error) {
+                this.log(`GET ID => ERROR ${error.message}`)
+                res.status(500).json(createError(error.message || 'Unknown error'))
+            }
+        })
+
+        api.post('/reviews', async (req, res) => {
+            const publish = get(req, 'query.publish', false) === 'true'
+            const expectedMultihash = get(req, 'query.expectedMultihash', null)
+            const bitcoinTransactionHash = get(req, 'query.bitcoinTransactionHash', null)
+            this.log(`Storing Review Record, publish: ${publish ? 'yes' : 'no'}, bitcoinTransactionHash: ${bitcoinTransactionHash}`)
+            try {
+                const reviewRecord = req.body
+                const issuer = get(reviewRecord, 'issuer')
+                await this.chluIpfs.waitUntilReady()
+                const did = this.chluIpfs.didIpfsHelper.didId
+                // TODO: HUGE BUG TO FIX
+                // chluIpfs keeps track of the last_reviewrecord_multihash but the api client does not
+                // know this and does not include it in the review record. The api client's signature
+                // will be invalidated.
+                const result = await this.chluIpfs.storeReviewRecord(reviewRecord, {
+                    signAsCustomer: false, // VERY IMPORTANT
+                    signAsIssuer: issuer === did, // Also very important, only sign as issuer if authorized to do so
```

Comment:
I think we should definitely make the publish method only publish crawled reviews. So instead of checking issuer === did, we should check if review is backed by popr, if yes, then don't publish it and return error. Does it make sense?

Reply:
Not really, I'd like a separate API for crawled reviews, because this one takes one review at a time. If the user imports 100 upwork reviews, we don't want 100 http calls.

We discussed this flow:

- user sends request to crawl
- backend crawls, sends result to user
- user sends accept request to backend by signing
- backend publishes reviews

I think we should leave the comment in but reword it (`HUGE BUG TO FIX` is not very professional :laughing: sorry about that) but then disable this API endpoint.

What do you think?

Reply:
But then, we can potentially remove this API?

Reply:
Yes, we don't need this anymore. I will remove it, but let's not squash the commits so that we can get it back if we need it